File: packages/client/src/app/classes/player/player.ts
import { PlayerData } from '@app/classes/communication/';
import { Tile } from '@app/classes/tile';
import { PublicUser } from '@common/models/user';
export default class Player {
    id: string;
    publicUser: PublicUser;
    score: number;
    private tiles: Tile[];

    constructor(id: string, publicUser: PublicUser, tiles: Tile[]) {
        this.id = id;
        this.publicUser = publicUser;
        this.score = 0;
        this.tiles = [...tiles];
    }

    getTiles(): Tile[] {
        return [...this.tiles];
    }

    updatePlayerData(playerData: PlayerData): void {
        this.id = playerData.newId ? playerData.newId : this.id;
        this.publicUser = playerData.publicUser ? playerData.publicUser : this.publicUser;
        this.score = playerData.score ? playerData.score : this.score;
        this.tiles = playerData.tiles ? [...playerData.tiles] : this.tiles;
    }
}

File: packages/mobile/lib/classes/player/player.dart
import 'package:mobile/classes/tile/tile.dart';
import 'package:mobile/classes/user.dart';

class PlayerData {
  String id;
  String? newId;
  PublicUser? publicUser;
  int? score;
  List<Tile>? tiles;

  PlayerData({
    required this.id,
    this.newId,
    this.publicUser,
    this.score,
    this.tiles,
  });
}
